docs/rules/no-whitespace-before-property.md

```diff
@@ -0,0 +1,53 @@
+# Disallow whitespace before properties (no-whitespace-before-property)
+
+Disallows whitespace around dot or brackets before properties of objects if they are on the same line.
```

Comment:
Please add more here. This section should explain the comfort that this rule is treating, not the rule itself. See http://eslint.org/docs/rules/comma-dangle as an example.